crates/services/p2p/src/p2p_service.rs

```diff
@@ -211,7 +213,8 @@ impl FuelP2PService {
     pub async fn new(
         reserved_peers_updates: broadcast::Sender<usize>,
         config: Config,
-        codec: PostcardCodec,
+        gossipsub_codec: UnboundedCodec<PostcardDataFormat>,
+        request_response_codec: BoundedCodec<PostcardDataFormat>,
```

Comment:
Given that these are the only two instances of using these codecs, perhaps we could simplify the implementation with more specific traits. What if we just renamed `UnboundedCodec` -> `GossipsubCodec` (perhaps we only need one trait and not two traits here), and renamed `BoundedCodec` -> `RequestResponseReadWrite` (or something similar)?

Reply:
Followed the suggestion: c5aa3622bb56ef69ce7aee44231ae20a9d05d2c3 and ead761a08a1a103bdd103f8f7e66da077fb92186